api/common.js

```diff
@@ -413,6 +414,57 @@ exports.load_github_detail = function(service_name, cb) {
     }).catch(cb)
 }
 
+
+exports.generateQuery = function(str){
+    let result=stopwords.cleanText(str).split(' ').filter(e => String(e).trim()).map((word) => `W='${word}'`).join(',');
+    return "OR("+result+")"
+}
+
+exports.updateProjectMag = function(project){
```

Comment:
you should add callback from mag.js and call it when it finishes processing one project. mag.js is currently using forEach, which means handleProject() is call all at once. Switch to use async.eachSeries, or arync.eachLimit() instead to limit the concurrency.